packages/babylon/src/parser/statement.js

```diff
@@ -1408,6 +1408,16 @@ export default class StatementParser extends ExpressionParser {
     } else if (this.match(tt.at)) {
       this.parseDecorators(false);
       return this.parseClass(expr, true, true);
+    } else if (
+      this.match(tt._let) ||
+      this.match(tt._const) ||
+      this.match(tt._var)
+    ) {
+      this.raise(
+        this.state.start,
+        "The `default` export can only take an expression;" +
```

Comment:
It can also take function and class declarations.

Reply:
So would you suggest to just add both to the error message?

Reply:
Maybe just:

> Only expressions, functions or class declarations are allowed as the default export.

(Let's add tests for those too).